scripts/create_cu_vm.py

```diff
@@ -0,0 +1,121 @@
+from utils.virtual_machines import deploy_template
+
+
+def command_run(ssh_client, command, message):
+    exit_status, output = ssh_client.run_command(command)
+    if exit_status != 0:
+        logger.error(message)
+        sys.exit(127)
+
+
+def make_ssh_client(ip, sshname, sshpass):
+    connect_kwargs = {
+        'username': sshname,
+        'password': sshpass,
+        'hostname': ip
+    }
+    return SSHClient(**connect_kwargs)
+
+
+def config_cu_vm(ssh_client):
+    """Downloads cu scripts and sets up cron jobs using ssh_client
+
+    Args:
+        ssh_client: :py:class:`utils.ssh.SSHClient` instance
+    """
+    url = cfme_data['basic_info']['capandu_scripts_url']
+
+    logger.info('Setting up cron jobs on the CU VM')
+
+    command_run(ssh_client, "cd /etc/init.d; wget {}/cu-disk-script.sh".format(url),
+        "CU: There was an error downloading disk script file")
+    command_run(ssh_client, "cd /etc/init.d; wget {}/cu-network-script.sh".format(url),
+        "CU: There was an error downloading network script file")
+    command_run(ssh_client,
+        "chmod +x /etc/init.d/cu-disk-script.sh /etc/init.d/cu-network-script.sh",
+        "CU: There was an error running chmod")
+    command_run(ssh_client, "cd /tmp; wget {}/crontab.in".format(url),
+        "CU: There was an error downloading crontab.in")
+    command_run(ssh_client, "crontab /tmp/crontab.in",
+        "CU: There was an error running crontab:")
+    command_run(ssh_client, "yum install -y iperf",
+        "CU: There was an error installing iperf")
+    command_run(ssh_client, "yum install -y pv",
+        "CU: There was an error installing pv")
+
+    # The cron jobs run at reboot.Hence,a reboot is required.
+    logger.info('Rebooting vm after setting up cron jobs')
+    command_run(ssh_client, "reboot &",
+        "CU: There was an error running reboot")
+
+
+def vm_running(provider, vm_name):
+    if provider.is_vm_running(vm_name):
+        logger.info("VM %s is running" % vm_name)
+    else:
+        logger.error("VM is not running")
+        return 10
+
+
+def cu_vm(provider, vm_name, template):
+    """
+    Deploys CU VM
+    """
+    # deploy_template(provider, vm_name, template, purpose)
+    provider_dict = cfme_data['management_systems'][provider]
+    datastore = provider_dict['cap_and_util']['allowed_datastores']
+    resource_pool = provider_dict['cap_and_util']['resource_pool']
+
+    deploy_template(provider, vm_name, template,
+        resourcepool=resource_pool, allowed_datastores=datastore)
+
+    prov_mgmt = get_mgmt(provider)
+    vm_running(prov_mgmt, vm_name)
+    ip = prov_mgmt.get_ip_address(vm_name)
+
+    vm_ssh_creds = provider_dict['capandu_vm_creds']
+    sshname = credentials[vm_ssh_creds]['username']
+    sshpass = credentials[vm_ssh_creds]['password']
+
+    # Create cron jobs to generate disk and network activity on the CU VM.
+    ssh_client = make_ssh_client(ip, sshname, sshpass)
+    config_cu_vm(ssh_client)
```

Comment:
This should be enclosed inside try/except block to delete the VM in case of any error. I see the `command_run` now exits using the exit function. I'd prefer to raise some exception (a custom class) and catching that exception down there in the `__main__` condition. Then it can turn those exceptions to error codes with error messages. Currently it calls the `sys.exit` function always with `None`, so currently it has no point. If you have these exceptions raised, you can just delete the VM in the except block and do `raise` to let it bubble up the stack till it is caught by the try/except enclosing the `main()` call.